### PR TITLE
v (closes #24)

### LINKED ISSUE #24: [enhancement] add the original author and message in the message sent by the bot when the original message is deleted
currently if the message the bot is trying to reply to doesn't exist you simply just send the message

it would be better to add the author's username and message in a quote block if the original message was deleted before the bot could reply, to avoid confusion

a format like this is good:

```
> [username]: [message]
[message generated by spongechat]
```

***\*\*\*(MAKE SURE TO PUT A `>` AT THE START OF EACH NEW LINE IN THE MESSAGE OR ELSE FORMATTING WOULD BREAK!!!!!)\*\*\****

or if you're too lazy to do that:

````
```
[username]: [message]
```
[message generated by spongechat]
````

### PR DESCRIPTION
**Pull request by Butterroach**
i coded this in github and i cant test this and this is a combination of the discord.js docs and a random stackoverflow answer

### CODE REVIEW

**Review by daniwasonline — DISMISSED** [review]
must mention skibidi toilet rizz and tiktok rizz party

File: index.js [hunk 1]
```diff
@@ -89,6 +89,9 @@ let lastMessage = "";
 let enableLocal = false;
 
 client.on("messageCreate", async message => {
+  let content = message.content
```
Comment (artifishvr) [inline index.js:92]: why? (92-94)

File: index.js [hunk 2]
```diff
@@ -140,9 +143,19 @@ client.on("messageCreate", async message => {
...
+    
     // Send AI response
-    message.reply({ content: `${response}`, failIfNotExists: false });
+    let no_longer_exists = false;
+    try {
+      message.channel.messages.fetch(message.id);
+    } except(e) {
+      console.warn(e);
+      no_longer_exists = true;
+      message.reply({ content: `\`\`\`\n${author_un}: ${content}\n\`\`\`\n\n${response}`, failIfNotExists: false });
```
Comment (artifishvr) [inline index.js:154]: yeah visually looks like it would work

File: index.js (hunk 1 above)
Comment (Butterroach) [reply index.js:92]: YUMMY VARIABLES

**Review by artifishvr — APPROVED** [review]

File: index.js [hunk 3]
```diff
@@ -142,7 +145,17 @@ client.on("messageCreate", async message => {
     }
 
     // Send AI response
-    message.reply({ content: `${response}`, failIfNotExists: false });
+    let no_longer_exists = false;
+    try {
+      message.channel.messages.fetch(message.id);
+    } except(e) {
```
Comment (artifishvr) [inline index.js:151]: dude...
```suggestion
    } catch(e) {
```

File: index.js [hunk 4]
```diff
@@ -89,6 +89,9 @@ let lastMessage = "";
 let enableLocal = false;
 
 client.on("messageCreate", async message => {
+  let content = message.content
+  let msg_id = message.id;
```
Comment (artifishvr) [inline index.js:93]: you dont even use this var

**Comment by daniwasonline** [comment]
js is so weird

File: index.js [hunk 5]
```diff
@@ -115,7 +115,7 @@ client.on("messageCreate", async message => {
     let formattedUserMessage = `${message.author.username} (${await getPronouns(message.author.id)}) on ${DateTime.now().setZone('utc').toLocaleString(DateTime.DATETIME_FULL)}: ${message.content}\n${imageDetails}`;
     if (message.reference) {
       await message.fetchReference().then(async (reply) => {
-        if (reply.author.id == "954288870244114473") {
+        if (reply.author.id == "954288870244114473") {  // FIXME
```
Comment (artifishvr) [inline index.js:118]: what?

Comment (Butterroach) [reply index.js:118]: make that not static

Comment (artifishvr) [reply index.js:118]: i cant remember what that's for?

Comment (Butterroach) [reply index.js:118]: #35

Comment (artifishvr) [reply index.js:118]: OH wait why tf is it hardcoded

Comment (Butterroach) [reply index.js:118]: laziness

Comment (Butterroach) [reply index.js:118]: image_proxy

**Review by artifishvr — APPROVED** [review]

**Pull request merged**